lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -101,7 +112,14 @@ def add_extra_data(post, options)
       end
 
       def add_shopper_interaction(post, payment, options={})
-        shopper_interaction = payment.verification_value ?  "Ecommerce" : "ContAuth"
+        if payment.is_a?(String)
```

Comment:
Was there a reason to break this out?

Reply:
`payment.verification_value` errors when `payment` is a string, so I restructured it this way. There is probably a more concise formulation though... hmm.

Reply:
The alternatives I've come up with are just less readable, for not much benefit. I'll go with this unless you have a specific suggestion.